### PR TITLE
Fix ListArtifacts and get-artifact compatibility errors with proxied artifacts

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Fix ListArtifacts and get-artifact compatibility errors with proxied artifacts. Addresses #5377.

## How is this patch tested?

- Unit tests
- Manual tests conducted via the MLflow UI (artifact listing and downloading from the artifact viewer for HTTP, `mlflow-artifacts`, and non-proxied - e.g. `/tmp/mylocaldirectory` routes)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

## Release Notes

Fix listing and downloading of artifacts from the MLflow UI when the MLflow server is operating in `--serve-artifacts` mode.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by dbczumar** [comment]
autoformat

File: tests/tracking/test_mlflow_artifacts.py [hunk 1]
```diff
@@ -245,3 +245,58 @@ def test_mlflow_artifacts_example(tmpdir):
...
+        if path:
+            resp = requests.get(url, params={"run_id": run_id, "path": path})
+        else:
+            resp = requests.get(url, params={"run_id": run_id})
+        resp.raise_for_status()
+        return resp.json()
+
+    url = artifacts_server.url
+    mlflow.set_tracking_uri(url)
+    api = f"{url}/api/2.0/mlflow/artifacts/list"
```
Comment (eedeleon) [inline tests/tracking/test_mlflow_artifacts.py:261]: Given the MLflow Tracking URI tends to be the base for the artifact URI(url is passed through here for that reason)

Is there a way to deploy a proxied artifact server with artifact uri mlflow-artifacts, that gets red client side into
Tracking UI + /api/2.0/mlflow/artifacts

Effectively, can we make this short cut you did for testing a supprted feature in the client + mlflow server call?

**Comment by dbczumar** [comment]
autoformat

File: tests/tracking/test_mlflow_artifacts.py (hunk 1 above)
Comment (dbczumar) [reply tests/tracking/test_mlflow_artifacts.py:261]: Hi Eddie! Thanks for asking about this - we considered using `api/2.0/mlflow/artifacts` as the base but opted to go with `mlflow-artifacts` to distinguish the API surface for proxied artifact uploads from the API surface for MLflow Tracking. If we included these in the MLflow Tracking surface, all implementers of MLflow Tracking would be required to support it.

**Review by BenWilson2 — APPROVED** [review]
Looks great! The docstrings are amazingly clear and will help out substantially if we ever need to touch this logic again.

**Pull request merged**